Remove direct installation instructions, link to docs

slhck opened the pull request:
Change default installation instructions to remove use of "sudo", as it is highly discouraged.

References:

- https://pages.charlesreid1.com/dont-sudo-pip/
- https://askubuntu.com/questions/802544/is-sudo-pip-install-still-a-broken-practice
- https://stackoverflow.com/questions/21055859/what-are-the-risks-of-running-sudo-pip

Other very popular packages also refraining from using "sudo" in their documentation:

- https://pyyaml.org/wiki/PyYAMLDocumentation
- https://github.com/psf/requests
- https://docs.scipy.org/doc/numpy/user/building.html

---

**Edit:** PR changed according to feedback.

Uinden commented:
Isn't it `pip install streamlink --user`?
Without `--user` it will fail. At least on my installation. 
Like ```error: could not create '/usr/lib/python3.7/site-packages/%package%': Permission denied```

slhck commented:
It depends!™️

If you're using a user-environment Python like from `pyenv` or Homebrew, or Pipenv, or a Virtualenv, then you don't need `--user`. On a “standard” Linux with system Python, you need `--user`.

If there's a danger that people might just do `sudo` if they see a “permission denied” error, then perhaps we should add `--user`. It doesn't hurt.

bastimeyer commented:
The install instructions on the repo's README are a remnant of the Livestreamer fork, I believe. Streamlink's documentation has a dedicated install page, where the `pip install --user` situation is explained, although kind of mixed with installing it from git, see here:
https://streamlink.github.io/install.html#source-code

Not sure if everyone else agrees with me here, but I think it would be better to remove the install commands from the README as a whole and just refer to the install documentation instead. Ideally, this should have been done when the the install page and Github release template got rewritten earlier this year, but I forgot about it.

I won't close any PRs this month due to Hacktoberfest, but if we agree on removing the commands, I would be happy if you could change this PR real quick. If not, then just removing the sudo part is also fine.

gravyboat commented:
@bastimeyer I agree. I think a link to the actual install docs would be better, then we don't have to maintain them in two locations.

slhck commented:
Thanks for your feedback everyone. I've changed the README to just link to the documentation for installation instructions. PR title changed accordingly.

slhck commented:
You're right. I fixed those links, removed the explicit mention of Windows portable (since it's linked from the first link anyways), added macOS and the source code installation.

gravyboat commented:
Looks good, thanks @slhck!
